storage/src/block_meta_storage.rs

```diff
@@ -401,13 +401,6 @@ impl Decoder for Meta {
             // predecessor
             let predecessor = if has_predecessor!(mask) {
                 let block_hash = bytes[IDX_PREDECESSOR..IDX_LEVEL].to_vec();
-                assert_eq!(
```

Comment:
This should newer be triggered as the resulting length is  `IDX_LEVEL-IDX_PREDECESSOR` that is `LEN_BLOCK_HASH`, and also correct length is checked below in `block_hash.try_into()?`.